SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/repository/SubscriptionInfoDAO.java

```diff
@@ -1,27 +1,280 @@
+    }
+
+    public Uni<RowSet<Row>> deleteSubscription(DeleteSubscriptionRequest request) {
+        return clientManager.getClient(request.getTenant(), false).onItem().transformToUni(client -> client.preparedQuery("DELETE FROM subscriptions WHERE subscription_id=$1")
+                .execute(Tuple.of(request.getId())));
+    }
+
+    public Uni<RowSet<Row>> getAllSubscriptions(String tenant, int limit, int offset) {
+        return clientManager.getClient(tenant, false).onItem().transformToUni(client -> client.preparedQuery("SELECT subscription FROM subscriptions LIMIT $1 OFFSET $2")
+                .execute(Tuple.of(limit, offset)).onFailure().retry().atMost(3));
+    }
+
+    public Uni<RowSet<Row>> getSubscription(String tenant, String subscriptionId) {
+        return clientManager.getClient(tenant, false).onItem().transformToUni(client -> client.preparedQuery("SELECT subscription FROM subscriptions WHERE subscription_id=$1")
+                .execute(Tuple.of(subscriptionId)).onFailure().retry().atMost(3));
+    }
+
+    public Uni<RowSet<Row>> getRegById(String tenant, String id) {
+        return clientManager.getClient(tenant, false).onItem().transformToUni(client -> client.preparedQuery("SELECT reg FROM csource WHERE id = $1").execute(Tuple.of(id)).onFailure()
+                .retry().atMost(3));
+    }
+
+    public Uni<Void> updateNotificationSuccess(String tenant, String id, String date) {
+        return clientManager.getClient(tenant, false).onItem().transformToUni(client -> {
+           return client
+                    .preparedQuery("UPDATE subscriptions SET subscription = subscription || ('{\""
+                            + NGSIConstants.NGSI_LD_TIMES_SENT + "\": [{\"" + NGSIConstants.JSON_LD_VALUE
+                            + "\": '|| (subscription#>>'{" + NGSIConstants.NGSI_LD_TIMES_SENT + ",0, "
+                            + NGSIConstants.JSON_LD_VALUE + "}')::integer + 1 ||'}],\""
+                            + NGSIConstants.NGSI_LD_LAST_SUCCESS + "\": [{\"" + NGSIConstants.JSON_LD_TYPE + "\": \""
+                            + NGSIConstants.NGSI_LD_DATE_TIME + "\", \"" + NGSIConstants.JSON_LD_VALUE + "\": \"$1\"}],\""
+                            + NGSIConstants.NGSI_LD_LAST_NOTIFICATION + "\": [{\"" + NGSIConstants.JSON_LD_TYPE
+                            + "\": \"" + NGSIConstants.NGSI_LD_DATE_TIME + "\", \"" + NGSIConstants.JSON_LD_VALUE
+                            + "\": \"$1\"}]}')::jsonb WHERE subscription_id=$2")
+                    .execute(Tuple.of(date, id)).onFailure().retry().atMost(3).onItem()
+                    .transformToUni(t -> Uni.createFrom().voidItem());
+        });
+    }
+
+    public Uni<Void> updateNotificationFailure(String tenant, String id, String date) {
+        return clientManager.getClient(tenant, false).onItem().transformToUni(client -> client
+                .preparedQuery("UPDATE subscriptions SET subscription = subscription || ('{\""
```

Comment:
It should not conflict with registry subscription. Please verfiy.

Reply:
done